### DISCUSSION TITLE
[@types/react] cannot return undefined in React.FC

### DISCUSSION THREAD

**Discussion by vimutti77** [DefinitelyTyped/DefinitelyTyped#60242] [Issues with a @types package] [answered]
Now, React 18 is [Allow components to render undefined](https://github.com/facebook/react/pull/21869)

But when I return undefined in component
```
const MyComponent: FC = () => {
  return undefined
}
```
I get this error
```
Type '() => undefined' is not assignable to type 'FC<{}>'.
  Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.ts(2322)
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
This is essentially a duplicate of #18912 i.e. it's not an issue with `@types/react` but `typescript`. Once  Microsoft/TypeScript#21699 is resolved we should be able to use `ReactNode` as a valid return type for function components which includes `string` and `boolean`.
